Exiled.Events/Patches/Events/Cassie/SendingCassieMessage.cs

```diff
@@ -7,66 +7,59 @@
 
             var returnLabel = generator.DefineLabel();
 
-            // SendingCassieMessageEventArgs ev = new SendingCassieMessageEventArgs(words, makeHold, makeNoise, true);
-            // Cassie.OnSendingCassieMessage(ev);
-            //
-            // words = ev.Words;
-            // makeHold = ev.MakeHold;
-            // makeNoise = ev.MakeNoise;
-            //
-            // if (!ev.IsAllowed)
-            //     return;
-            newInstructions.InsertRange(0, new[]
+            newInstructions.InsertRange(index, new[]
             {
-                new CodeInstruction(OpCodes.Ldarg_0),
+                // string words
+                new CodeInstruction(OpCodes.Ldarg_0).MoveLabelsFrom(newInstructions[index]),
+
+                // bool makeHold
                 new CodeInstruction(OpCodes.Ldarg_1),
+
+                // bool makeNoise
                 new CodeInstruction(OpCodes.Ldarg_2),
+
+                // var ev = SendingCassieMessageEventArgs(...)
                 new CodeInstruction(OpCodes.Ldc_I4_1),
                 new CodeInstruction(OpCodes.Newobj, GetDeclaredConstructors(typeof(SendingCassieMessageEventArgs))[0]),
                 new CodeInstruction(OpCodes.Dup),
-                new CodeInstruction(OpCodes.Dup),
-                new CodeInstruction(OpCodes.Dup),
-                new CodeInstruction(OpCodes.Dup),
-                new CodeInstruction(OpCodes.Call, Method(typeof(Handlers.Cassie), nameof(Handlers.Cassie.OnSendingCassieMessage))),
-                new CodeInstruction(OpCodes.Call, PropertyGetter(typeof(SendingCassieMessageEventArgs), nameof(SendingCassieMessageEventArgs.Words))),
-                new CodeInstruction(OpCodes.Starg_S, 0),
-                new CodeInstruction(OpCodes.Call, PropertyGetter(typeof(SendingCassieMessageEventArgs), nameof(SendingCassieMessageEventArgs.MakeHold))),
-                new CodeInstruction(OpCodes.Starg_S, 1),
-                new CodeInstruction(OpCodes.Call, PropertyGetter(typeof(SendingCassieMessageEventArgs), nameof(SendingCassieMessageEventArgs.MakeNoise))),
-                new CodeInstruction(OpCodes.Starg_S, 2),
+
+                // Cassie.SendingCassieMessageEventArgs(ev);
+                new CodeInstruction(OpCodes.Call, Method(typeof(Cassie), nameof(Cassie.OnSendingCassieMessage))),
+
+                // if (!ev.IsAllowed) return;
                 new CodeInstruction(OpCodes.Callvirt, PropertyGetter(typeof(SendingCassieMessageEventArgs), nameof(SendingCassieMessageEventArgs.IsAllowed))),
                 new CodeInstruction(OpCodes.Brfalse_S, returnLabel),
             });
 
             newInstructions[newInstructions.Count - 1].WithLabels(returnLabel);
 
-            for (int i = 0; i < newInstructions.Count; i++)
-                yield return newInstructions[i];
+            for (var z = 0; z < newInstructions.Count; z++)
+                yield return newInstructions[z];
```

Comment:
Why'd you make it `z`?

Reply:
because z is what we use for all patches

Reply:
Ok that's gross but fine.